services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/DataPlaneProcessor.java

```diff
@@ -229,6 +230,38 @@ private void setTheMissingFields(PortContext context, List<PortEntity> portEntit
 
     }
 
+    private List<ResourceOperation> initializeResourceOperationTypes(PortContext context, List<PortEntity> portEntities) {
+        List<ResourceOperation> resourceOperationTypes = new ArrayList<>();
+        resourceOperationTypes.add(new ResourceOperation(Common.ResourceType.PORT, Common.OperationType.CREATE));
+        resourceOperationTypes.add(new ResourceOperation(Common.ResourceType.NEIGHBOR, Common.OperationType.CREATE));
+
+        // TODO: to enable full sg support, we will need to add full security group entities to context and check context here
+        for (PortEntity port : portEntities) {
+            if (!CommonUtil.isNullOrEmpty(context.getDefaultSgId()) && port.getPortSecurityEnabled()) {
+                resourceOperationTypes.add(new ResourceOperation(Common.ResourceType.SECURITYGROUP, Common.OperationType.CREATE));
+                break;
+            }
+        }
+
+        if (context.containRouters()) {
+            resourceOperationTypes.add(new ResourceOperation(Common.ResourceType.NEIGHBOR, Common.OperationType.CREATE));
```

Comment:
Resrource type NEIHBOR was already added at line 236, this looks like a duplicate. From the context it looks like resource type ROUTER might be the correct one.
If this is in fact, the correct resource type, a clarifying comment will remove the confusion.